modules/engine-prototype/src/main/scala/engine/Sequence.scala

```diff
@@ -0,0 +1,91 @@
+package engine
+
+import scala.concurrent.ExecutionContext
+
+import cats.{FlatMap, Functor}
+import cats.data.NonEmptyList
+import cats.effect.Effect
+import cats.effect.implicits._
+import cats.implicits._
+
+import fs2.Stream
+import fs2.async
+import fs2.async.mutable.Signal
+
+import monocle.{Lens, Prism}
+import monocle.macros.GenLens
+
+sealed trait Sequence
+object Sequence {
+
+  sealed trait F2 extends Sequence
```

Comment:
I agree in general with the idea, but I think building the ADT like this will be long and tedious. Maybe we could declare `Sequence` as
```
sealed trait Sequence[I<:Resource.Instrument]
```
and then
```
object Sequence {
  type F2 = Sequence[Resource.Instrument.F2]
}
```